tests/testthat/test-combine-categories.R

```diff
@@ -169,4 +190,18 @@ with_test_authentication({
         expect_identical(names(categories(ds$combined_petloc)),
             c("Mammals", "Bird", "Skipped", "Not Asked"))
     })
+
+    test_that("collapseCategories works on categorical variable", {
+        ds$cat <- factor(sample(c("cat", "Cat", "dog", "aphid"), nrow(ds), replace = TRUE))
+        var <- ds$cat
+        var <- collapseCategories(var, c("cat", "Cat"), "cat")
+        expect_identical(names(categories(var)), c("aphid", "cat", "dog", "No Data"))
+        var <- collapseCategories(var, c("cat", "aphid"), "nope")
+        expect_identical(names(categories(var)), c("dog", "No Data", "nope"))
+        var <- collapseCategories(var, var %in% "nope", "yup")
```

Comment:
Oh there's the expression test. Now that I see it, I'm not sure I see the value in supporting this. Seems like it could blow up in unexpected/undesirable ways.